src/NuGet.Core/NuGet.Packaging/PackageExtraction/ZipArchiveExtensions.cs

```diff
@@ -85,5 +87,23 @@ public static void UpdateFileTimeFromEntry(this ZipArchiveEntry entry, string fi
                 }
             }
         }
+
+        public static void UpdateFilePermissionsFromEntry(this ZipArchiveEntry entry, string fileFullPath, ILogger logger)
+        {
+            if (RuntimeEnvironmentHelper.IsWindows)
+            {
+                return;
+            }
+            // Entry permissions are not restored to maintain backwards compatibility with .NET Core 1.x.
+            // (https://github.com/NuGet/Home/issues/4424)
+            // On .NET Core 1.x, all extracted files had default permissions of 766.
+            // The default on .NET Core 2.x has changed to 666.
+            // To avoid breaking executable files in existing packages (which don't have the x-bit set)
+            // we force the .NET Core 1.x default permissions.
+            chmod(fileFullPath, 0x1f6); // 0766
```

Comment:
This was set for all files. Not having the execute bit breaks native executables and scripts.